src/advanced/JupyterNotebook/index.tsx

```diff
@@ -0,0 +1,327 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+
+import React from 'react';
+import Grid from '../../layouts/Grid';
+import Container from '../../layouts/Container';
+import ReactMarkdown from 'react-markdown';
+import parse from 'html-react-parser';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { coy } from 'react-syntax-highlighter/dist/cjs/styles/prism';
+import { CodeComponent } from 'react-markdown/src/ast-to-react';
+import Heading from '../../components/Heading';
+import Paper from '../../layouts/Paper';
+
+/*
+ TODO:
+   - finesse table rendering
+   - centre images
+   - add worksheets handing option
+        - if has worksheets, use specified worksheet idx (0 if not specified)
+        - add worksheet idx parameter
+   - write documentation 
+*/
+
+// cats an array of lines together
+const sourceLines = (lines: any[]) => {
+    return lines.reduce((result, value) => {
+        return result + value;
+    }, '');
+};
+
+export interface JupyterNotebookProps {
+    notebookData: string;
+}
+
+interface JupyterNotebookCellProps {
+    cell: any;
```

Comment:
More specific type ?

Reply:
the cell comes from a JSON blob and there are variations of it out in the wild